camdecmpswks/functions/rpt_quartely_em_summary_data.sql

```diff
@@ -0,0 +1,310 @@
+    VOLATILE 
+    ROWS 1000
+    
+AS $BODY$
+DECLARE 
+    monLocIds character varying[];
+
+BEGIN
+    RETURN QUERY
+    (with lst as
+        (
+            select  pln.mon_plan_id,
+                    prd.rpt_period_id,
+                    mpl.mon_loc_id,
+                    pln.fac_id,
+                    prd.calendar_year,
+                    prd.quarter
+            from  camdecmpswks.monitor_plan pln,
+                    camdecmpswks.monitor_plan_location mpl,
+                    camdecmpsmd.reporting_period prd
+            where  pln.mon_plan_id = monplanid
+            and  mpl.mon_plan_id  = pln.mon_plan_id
+            and  (vyear is null or prd.calendar_year = vyear)
+        )
+        select
+                coalesce( unt.unitid, stp.stack_name) as location,
+                cmb.calendar_year,
+                cmb.Period,
+                cmb.Op_Hours,
+                cmb.Op_Days,
+                cmb.Op_Time,
+                cmb.Hit,
+                cmb.So2m,
+                cmb.Co2m,
+                cmb.Noxr,
+                cmb.Noxm,
+                cmb.Bco2
+        from  (
+                    --
+                    -- Quarterly Data for the Non-Fuel-Specific and Fuel-Specific Report Report Tables
+                    --
+                    select  sel.mon_plan_id,
+                            sel.rpt_period_id,
+                            sel.mon_loc_id,
+                            prd.calendar_year,
+                            ( 'QTR ' || prd.quarter ) as Period,
+                            null as fuel_cd,
+                            max( case when osd.op_type_cd = 'OPHOURS'   then osd.Op_Value end ) as Op_Hours,
+                            max( case when osd.op_type_cd = 'OPDAYS'    then osd.Op_Value end ) as Op_Days,
+                            max( case when osd.op_type_cd = 'OPTIME'    then osd.Op_Value end ) as Op_Time,
+                            max( case when osd.op_type_cd = 'HIT'       then osd.Op_Value end ) as Hit,
+                            max( case when osd.op_type_cd = 'SO2M'      then osd.Op_Value end ) as So2m,
+                            max( case when osd.op_type_cd = 'CO2M'      then osd.Op_Value end ) as Co2m,
+                            max( case when osd.op_type_cd = 'NOXR'      then osd.Op_Value end ) as Noxr,
+                            max( case when osd.op_type_cd = 'NOXM'      then osd.Op_Value end ) as Noxm,
+                            max( case when osd.op_type_cd = 'BCO2'      then osd.Op_Value end ) as Bco2
+                    from  (
+                                select  lst.mon_plan_id,
+                                        lst.rpt_period_id,
+                                        lst.mon_loc_id,
+                                        -- Indicates whether fuel specific data was reported for the location in the emission report
+                                        max(case when osd.fuel_cd is null then 0 else 1 end) as Fuel_Exists_Ind
```

Comment:
The only reason for this subquery is to get the Fuel_Exists_Ind value, but that is only needed for the fuel-specific results in the model query in [Model Query Comment](https://github.com/US-EPA-CAMD/easey-ui/issues/6111#issuecomment-2249198803).  Since this data is for the non-fuel-specific data, you should be able to replace the subquery with the "lst" with-query.

```
select  lst.mon_plan_id,
        lst.rpt_period_id,
        lst.mon_loc_id,
        prd.calendar_year,
        ( 'QTR ' || prd.quarter ) as Period,
        null as fuel_cd,
        max( case when osd.op_type_cd = 'OPHOURS'   then osd.Op_Value end ) as Op_Hours,
        max( case when osd.op_type_cd = 'OPDAYS'    then osd.Op_Value end ) as Op_Days,
        max( case when osd.op_type_cd = 'OPTIME'    then osd.Op_Value end ) as Op_Time,
        max( case when osd.op_type_cd = 'HIT'       then osd.Op_Value end ) as Hit,
        max( case when osd.op_type_cd = 'SO2M'      then osd.Op_Value end ) as So2m,
        max( case when osd.op_type_cd = 'CO2M'      then osd.Op_Value end ) as Co2m,
        max( case when osd.op_type_cd = 'NOXR'      then osd.Op_Value end ) as Noxr,
        max( case when osd.op_type_cd = 'NOXM'      then osd.Op_Value end ) as Noxm,
        max( case when osd.op_type_cd = 'BCO2'      then osd.Op_Value end ) as Bco2
  from  lst
        join camdecmpsmd.reporting_period prd
          on prd.rpt_period_id = lst.rpt_period_id
        left join camdecmpswks.operating_supp_data osd
          on osd.mon_loc_id = lst.mon_loc_id
         and osd.rpt_period_id = lst.rpt_period_id
         and osd.op_type_cd in ( 'BCO2', 'CO2M', 'HIT', 'NOXM', 'NOXR', 'OPDAYS', 'OPHOURS', 'OPTIME', 'SO2M' )
         and osd.fuel_cd is null 
 group
    by  lst.mon_plan_id,
        lst.rpt_period_id,
        lst.mon_loc_id,
        prd.calendar_year,
        prd.quarter,
        osd.fuel_cd
```

Reply:
Note that Justin and I discussed this separately and decided to keep his version of the SQL because my suggested version had an issue with the results.

Reply:
Fuel_Exists_Ind is not needed.